### PR TITLE
run credscan and Polichek as its own pipeline ()

### PR DESCRIPTION
**Pull request by p3dr0rv**
The CredScan task is a repo wide assessment, it doesn't need to run as a task under each pipeline within a repo it can run once as its own pipeline...once per each repo.

Update credscan, policheck and publish results to v3

Related PR:
https://github.com/AzureAD/ad-accounts-for-android/pull/1609
https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/1617
https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/1449

Pipeline
https://identitydivision.visualstudio.com/Engineering/_build?definitionScope=%5CAndroid%5CPull%20Request%20Validations%5CCredScan

### CODE REVIEW

**Review by shahzaibj — APPROVED** [review]

**Comment by p3dr0rv** [comment]
image

Note: CredScan pipeline will fail, it is pointing to an old template once that  common is merged this will succeed

File: azure-pipelines/pull-request-validation/credscan.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  branches:
+    include:
+    - dev
+    - master
+    - release/*
+  batch: True
+
+jobs:
+- job: Phase_1
+  displayName: Phase 1
```
Comment (AdamBJohnsonx) [inline azure-pipelines/pull-request-validation/credscan.yml:15]: Can we give these better names?

Comment (p3dr0rv) [reply azure-pipelines/pull-request-validation/credscan.yml]: yes ca9469fc08f2be16b6dfdbc16aec87f26f93763b

File: azure-pipelines/templates/steps/credscan-policheck.yml [hunk 2]
```diff
@@ -1,27 +1,33 @@
 # File: credscan-policheck.yml
```
Comment (iamgusain) [inline azure-pipelines/templates/steps/credscan-policheck.yml:1]: >credscan-policheck [](http://example.com/codeflow?start=8&length=18)

maybe we can rename this as "security-compliance-check.yml"

File: azure-pipelines/templates/steps/credscan-policheck.yml [hunk 3]
```diff
@@ -1,27 +1,33 @@
...
+# https://docs.microsoft.com/en-us/azure/security/develop/yaml-configuration#credential-scanner-task
+# https://microsoft.sharepoint.com/sites/globalreadiness/SitePages/PoliCheck.aspx
```
Comment (iamgusain) [inline azure-pipelines/templates/steps/credscan-policheck.yml:3]: ># [](http://example.com/codeflow?start=0&length=1)

nit: You can also add a link to Component Governance docs as comment

File: azure-pipelines/pull-request-validation/credscan.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# File: azure-pipelines\pull-request-validation\credscan.yml
```
Comment (iamgusain) [inline azure-pipelines/pull-request-validation/credscan.yml:1]: >credscan [](http://example.com/codeflow?start=48&length=8)

nit: compliance-check.yml?

**Review by iamgusain — APPROVED** [review]
Approved with suggestions.

File: azure-pipelines/pull-request-validation/credscan.yml (hunk 4 above)
Comment (p3dr0rv) [reply azure-pipelines/pull-request-validation/credscan.yml]: change applied here 983cd5ca9f5381e5fcba8a6cb3ff3626253c293c

File: azure-pipelines/templates/steps/credscan-policheck.yml (hunk 2 above)
Comment (p3dr0rv) [reply azure-pipelines/templates/steps/credscan-policheck.yml]: change applied here 983cd5ca9f5381e5fcba8a6cb3ff3626253c293c

**Pull request merged**